scipy/optimize/tests/test_optimize.py

```diff
@@ -290,6 +290,19 @@ def test_powell_bounded(self):
             assert self.funccalls <= 131 + 20
             assert self.gradcalls == 0
 
+    def test_warn_neldermead(self):
+        with pytest.warns(UserWarning, match=r'Maximum number of function'):
+            func = lambda x: np.sum(x ** 2)
+            p0 = [0.15746215, 0.48087031, 0.44519198, 0.4223638, 0.61505159,
+                  0.32308456, 0.9692297, 0.4471682, 0.77411992, 0.80441652,
+                  0.35994957, 0.75487856, 0.99973421, 0.65063887, 0.09626474]
+            optimize.optimize._minimize_neldermead(func, p0, disp=True)
+
+        with pytest.warns(UserWarning, match=r'Maximum number of iterations'):
+            x0 = np.zeros(10)
+            func = optimize.rosen
+            optimize.fmin(func, x0, maxiter=5)
```

Comment:
Could you make this work with `_minimize_neldermead`? It would make more sense to call this instead.

Reply:
I agree with consistency, but actually, what do you think about having _both_ of these tests use the `minimize` interface, since we are interested in what users will see when they use the recommended public function?